### PR TITLE
Send options to through2 via dest(), src(), symlink()

### PR DESCRIPTION
**Pull request by vineethawal**
Ability to send options to through2.
Issue: default value set for highWaterMark in through2 is 16(max buffer length), when buffer length exceeds this default limit all files are not written to destination.

### CODE REVIEW

**Comment by sindresorhus** [comment]
:-1: That would only mask your actual problem. The max buffer is 16, true, but it should flush and continue after that, but something is stopping it. You should rather fix the actual problem.

**Comment by phated** [comment]
While I agree with @sindresorhus that the pipeline should be fixed (return the stream, call resume, etc).  I don't think it is wise for us to not allow a user to pass options to through2.  I thought this was already fixed in 2.x though...

**Comment by phated** [comment]
@vineethawal if we are going to support this, it needs to be uniform across the board, in `src`, `dest`, `symlink` (maybe even in glob-stream?).  We aren't __only__ going to support it in `dest`

**Comment by contra** [comment]
@phated Agree, we shouldn't be blocking the users from passing options to the underlying stream. Also, this won't even mask the problem because your plugins will also have a high water mark of 16

**Comment by vineethawal** [comment]
@phated Yes I agree, on it!

**Comment by phated** [comment]
@vineethawal looking good.  Please update the docs to mention that options are passed to through2 and add some tests.  Thanks

**Comment by vineethawal** [comment]
Added Documentation + test cases

**Comment by vineethawal** [comment]
@phated if the PR looks good can you have this merged?

**Comment by vineethawal** [comment]
@phated can we have this merged? waiting to use this feature in my project

**Comment by phated** [comment]
@vineethawal I was reviewing this but we have a bigger problem in issue #127 / #128 that needs to get in ASAP.  I also noticed that only a couple of the changes are being tested.

**Comment by phated** [comment]
@vineethawal looks like I messed up the merge with my docs update, could you rebase and get your docs updates to merge cleanly?  thanks!

**Comment by vineethawal** [comment]
@phated Done!

**Comment by phated** [comment]
Awesome. Thanks! I'm busy during the day today but I will take a look tonight

File: README.md [hunk 1]
```diff
@@ -224,6 +225,10 @@ Type: `Boolean` or `Object`
 
 Default: `undefined` (do not write sourcemaps)
 
+##### other
```
Comment (contra) [inline README.md]: I think we should pull the documentation from the through2 repo instead of linking

Comment (phated) [inline README.md]: I generally agree with surfacing docs but I don't think people should be using these options so I'm good with them not being surfaced.

**Pull request merged**